test/Groups/elements.jl

```diff
@@ -68,6 +68,39 @@ end
   end
 end
 
+@testset "Eltypes" begin
+   @test eltype(PermGroup)==PermGroupElem
+   @test eltype(PcGroup)==PcGroupElem
+   @test eltype(FPGroup)==FPGroupElem
+   @test eltype(MatrixGroup)==MatrixGroupElem
+   @test eltype(GL(2,3))==MatrixGroupElem{fq_nmod,fq_nmod_mat}
+   @test eltype(DirectProductGroup)==Oscar.BasicGAPGroupElem{DirectProductGroup}
+   @test eltype(direct_product(symmetric_group(3),cyclic_group(2)))==Oscar.BasicGAPGroupElem{DirectProductGroup}
+   @test eltype(SemidirectProductGroup)==Oscar.BasicGAPGroupElem{SemidirectProductGroup}
+   @test eltype(WreathProductGroup)==Oscar.BasicGAPGroupElem{WreathProductGroup}
+   @test eltype(AutomorphismGroup{PcGroup})==Oscar.BasicGAPGroupElem{AutomorphismGroup{PcGroup}}
+
+   G = symmetric_group(5)
+   x = cperm([1,4,2,5])
+   H = sub(G,[x])[1]
+   y = cperm(G,[2,3,4])
+   cc = conjugacy_class(G,y)
+   cs = conjugacy_class(G,H)
+   lc = x*H
+   @test [z for z in G] == @inferred collect(G)
+   @test [z for z in cc] == @inferred collect(cc)
+   @test [z for z in cs] == @inferred collect(cs)
+   @test [z for z in lc] == collect(lc)
```

Comment:
Why is there no `@inferred` here?

Reply:
The return type of `collect(lc)` is `Array` for `lc` group coset, so it returns an error. I don't know why this does not give any problem with conjugacy classes. The iterators are defined in the same way...

Reply:
That sounds as if the test uncovered a bug; we should fix the bug, not disable the test...

what type does `lc` have here? Is eltype implemented for that type?

Reply:
I solved the bug by adding the line `Base.IteratorSize(::Type{<:GroupCoset}) = Base.SizeUnknown()` before the implementation of the iterator for `GroupCoset`. But I have no idea why this influences the return type of `collect`.